lock.go

```diff
@@ -102,8 +121,9 @@ func (l *Locker) create() (bool, error) {
 	}
 
 	// Calculate the timestamp we are willing to wait for
-	stop := time.Now().Add(l.opts.WaitTimeout)
-	retries := l.opts.RetriesCount
+	deadline := time.Now().Add(l.opts.WaitTimeout)
+	var retryDelay *time.Timer
+
 	for {
```

Comment:
what about:
```go
var ticker = time.NewTicker(l.opts.WaitRetry)

for range ticker.C {
 		// ...do what you want 
}
```

Reply:
@dim what you think?

Reply:
1. that's not the same as a `WaitDelay` as it includes the time it takes to obtain the lock
2. it will wait for `l.opts.WaitRetry` before the first attempt
3. I don't like to allocate something when not needed, especially when the default setting is "no retry"

I will leave it as it is